app/controllers/application_controller.rb

```diff
@@ -1,10 +1,123 @@
 require './config/environment'
-
+require 'pry'
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+      enable :sessions
+      set :session_secret, "secret"
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if Helpers.is_logged_in?(session)
+      redirect '/tweets'
+    else
+      erb :signup
+    end
+  end
+
+  get '/login' do
+    if Helpers.is_logged_in?(session)
+      redirect '/tweets'
+    else
+      erb :login
+    end
+  end
+
+  post '/signup' do
+    if !params['username'].empty? && !params['email'].empty? && !params['password'].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.